Implement Leaves Blockdata

thelooter opened the pull request:
# Description

<!-- State the changes of the pull request below. -->

# Checklist

The following items should be checked before the pull request can be merged.

- [ ] Code follows existing style.
- [ ] Unit tests added (if applicable).

thelooter committed:
feat(blockdata): Implement Leaves Blockdata

Signed-off-by: thelooter <evekolb2204@gmail.com>

# Conflicts:
#	src/main/java/org/mockbukkit/mockbukkit/block/data/BlockDataKey.java

thelooter committed:
fix(blockdata): Unskip leaves test

Signed-off-by: thelooter <evekolb2204@gmail.com>

thelooter committed:
fix(blockdata): Correct distance validation and update Leaves initialization

- Fixed `setDistance` method to properly validate the distance range.
- Updated `BlockDataMockFactory` to include `Leaves` support.
- Adjusted test case to align with corrected distance logic.

thelooter committed:
fix(tests): Correct regex assertions in LeavesDataMockTest

- Fixed swapped comparison messages in `setDistance` exception tests.
- Minor code style adjustments in `LeavesDataMock`.

thelooter committed:
fix(tests): Update exception messages in LeavesDataMockTest

- Replaced regex assertions with exact message matching for distance validation exceptions.

thelooter committed:
refactor(tests): Remove unused `matchesRegex` import from LeavesDataMockTest

4everTheOne commented on src/main/java/org/mockbukkit/mockbukkit/block/data/LeavesDataMock.java:
```diff
@@ -0,0 +1,84 @@
+{
+
+	private static final int MAX_DISTANCE = 7;
+	private static final int MIN_DISTANCE = 1;
```
This values can be mined using MetaMiner. For example see the `TurleEgg` in `MaterialDataGenerator` :
https://github.com/MockBukkit/MockBukkit/blob/af34b1d30cbc32ff2055b07a72e379fe74aec96f/metaminer/src/main/java/org/mockbukkit/metaminer/internal/MaterialDataGenerator.java#L153-L157

thelooter committed:
feat(blockdata): Add dynamic distance limit support for Leaves

- Replaced static distance limits with dynamic values fetched from `BlockDataLimitation`.
- Updated `LeavesDataMock` logic to utilize the new dynamic limits.
- Enhanced material data to include `maxDistance` and `minDistance` properties for all leaf types.
- Adjusted material data generator to serialize `Leaves` distance limits.

Signed-off-by: thelooter <evekolb2204@gmail.com>
